CreateUser/handler.ts

```diff
@@ -0,0 +1,192 @@
+): TaskEither<Error, GraphRbacManagementClient> {
+  return tryCatch(
+    () =>
+      msRestNodeAuth.loginWithServicePrincipalSecret(
+        adb2cCreds.clientId,
+        adb2cCreds.secret,
+        adb2cCreds.tenantId,
+        { tokenAudience: "graph" }
+      ),
+    toError
+  ).map(
+    credentials =>
+      // the below cast to `any` is needed to prevent build errors
+      // due to some type mismatches among the azure-sdk-for-js libraries
+      // @see: https://github.com/Azure/azure-sdk-for-js/issues/7258
+      // tslint:disable-next-line:no-any
+      new GraphRbacManagementClient(credentials as any, adb2cCreds.tenantId)
+  );
+}
+
+export function CreateUserHandler(
+  adb2cCredentials: IServicePrincipalCreds,
+  apimCredentials: IServicePrincipalCreds,
+  azureApimConfig: IAzureApimConfig
+): ICreateUserHandler {
+  return async (context, _, __, ___, userPayload) => {
+    const internalErrorHandler = (errorMessage: string, error: Error) =>
+      genericInternalErrorHandler(
+        context,
+        "CreateUser | " + errorMessage,
+        error,
+        errorMessage
+      );
+    const response = await getGraphRbacManagementClient(adb2cCredentials)
+      .mapLeft(error =>
+        internalErrorHandler("Could not get the ADB2C client", error)
+      )
+      .chain(graphRbacManagementClient =>
+        tryCatch(
+          () =>
+            graphRbacManagementClient.users.create({
+              accountEnabled: true,
+              creationType: "LocalAccount",
+              displayName: `${userPayload.first_name} ${userPayload.last_name}`,
+              givenName: userPayload.first_name,
+              mailNickname: userPayload.email.split("@")[0],
+              passwordProfile: {
+                forceChangePasswordNextLogin: true,
+                password: randomString.generate({ length: 32 })
+              },
+              signInNames: [
+                {
+                  type: "emailAddress",
+                  value: userPayload.email
+                }
+              ],
+              surname: userPayload.last_name,
+              userPrincipalName: `${ulid()}@${adb2cCredentials.tenantId}`,
+              userType: "Member"
+            }),
+          toError
+        ).mapLeft(error =>
+          internalErrorHandler("Could not create the user on the ADB2C", error)
+        )
+      )
+      .chain(userCreateResponse =>
+        getApiClient(apimCredentials, azureApimConfig.subscriptionId)
+          .mapLeft(error =>
+            internalErrorHandler("Could not get the APIM client", error)
+          )
+          .map(apimClient => ({
+            apimClient,
+            userId: userCreateResponse.objectId
+          }))
+      )
+      .chain(taskResults =>
+        tryCatch(
+          () =>
+            taskResults.apimClient.user.createOrUpdate(
+              azureApimConfig.apimResourceGroup,
+              azureApimConfig.apim,
+              taskResults.userId,
+              {
```

Comment:
you must link the adb2c identity to the one generated into the adb2c, see this code:

https://github.com/gunzip/adb2c-apim-migrate/blob/master/migrate.ts#L280